sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/SymbolicExecution/ExplodedGraphTests.cs

```diff
@@ -222,39 +159,21 @@ public void ExplodedGraph_SingleBranchVisited_If()
                     }
                 };
 
-            explodedGraph.Walk();
+            context.Walk(8);
 
-            explorationEnded.Should().BeTrue();
-            numberOfProcessedInstructions.Should().Be(8);
-            numberOfExitBlockReached.Should().Be(1);
             numberOfLastInstructionVisits.Should().Be(1);
         }
 
         [TestMethod]
         [TestCategory("Symbolic execution")]
         public void ExplodedGraph_SingleBranchVisited_And()
         {
-            var testInput = "var a = false; if (a && !a) { a = true; }";
-            var method = ControlFlowGraphTest.CompileWithMethodBody(string.Format(TestInput, testInput), "Bar", out var semanticModel);
-            var methodSymbol = semanticModel.GetDeclaredSymbol(method);
-            var aSymbol = GetSymbol(semanticModel, method, "a");
-
-            var cfg = CSharpControlFlowGraph.Create(method.Body, semanticModel);
-            var lva = CSharpLiveVariableAnalysis.Analyze(cfg, methodSymbol, semanticModel);
-
-            var explodedGraph = new CSharpExplodedGraph(cfg, methodSymbol, semanticModel, lva);
-            var explorationEnded = false;
-            explodedGraph.ExplorationEnded += (sender, args) => { explorationEnded = true; };
-
-            var numberOfExitBlockReached = 0;
-            explodedGraph.ExitBlockReached += (sender, args) => { numberOfExitBlockReached++; };
-
-            var numberOfProcessedInstructions = 0;
-
-            explodedGraph.InstructionProcessed +=
+            const string testInput = "var a = false; if (a && !a) { a = true; }";
+            var context = new ExplodedGraphContext(testInput);
+            var aSymbol = context.GetSymbol("a");
+            context.EG.InstructionProcessed +=
                 (sender, args) =>
                 {
-                    numberOfProcessedInstructions++;
                     if (args.Instruction.ToString() == "a = !true")
                     {
                         args.ProgramState.GetSymbolValue(aSymbol).Should().Be(SymbolicValue.False); // Roslyn is clever !true has const value.
```

Comment:
Are we actually visiting this check here ? Looking at the input code now, the instruction `a = !true` is not present there

Reply:
Good catch! It has been there longer than 2 years already 😄